commit ebf4043216afacc206a845614d7e97be31082ef1
Author: Fabio Erculiani <lxnay@sabayon.org>
Date:   Wed Feb 9 17:46:55 2011 +0100

    entropy: do not switch UrlFetcher using a Client property (forward compatibility)

diff --git a/backends/entropy/entropyBackend.py b/backends/entropy/entropyBackend.py
index 85e97c3..61d6e82 100755
--- a/backends/entropy/entropyBackend.py
+++ b/backends/entropy/entropyBackend.py
@@ -701,6 +701,8 @@ class PackageKitEntropyClient(Client):
 
     _pk_progress = None
     _pk_message = None
+    def init_singleton(self):
+        Client.init_singleton(self, url_fetcher = PkUrlFetcher)
 
     def output(self, text, header = "", footer = "", back = False,
         importance = 0, level = "info", count = None, percent = False):
@@ -806,7 +808,6 @@ class PackageKitEntropyBackend(PackageKitBaseBackend, PackageKitEntropyMixin):
         self.doLock()
         signal.signal(signal.SIGQUIT, self.__sigquit)
         PkUrlFetcher._pk_progress = self.sub_percentage
-        self._entropy.urlFetcher = PkUrlFetcher
         self._repo_name_cache = {}
         PackageKitEntropyClient._pk_progress = self.percentage
         PackageKitEntropyClient._pk_message = self._generic_message
